### PR TITLE
Add Anthropic API tests

### PR DESCRIPTION
**Pull request by rchan26**
Towards #79 
- Add codecov and update README
- Added Anthropic tests
- Fixed a few bugs that were in Anthropic
- Fixed a few bugs in other models that were in Anthropic

### CODE REVIEW

File: tests/apis/anthropic/test_anthropic.py [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+        Warning, match="Environment variable 'ANTHROPIC_API_KEY' is not set"
+    ):
+        raise test_case[0]
+
+    # no warnings or errors
+    monkeypatch.setenv("ANTHROPIC_API_KEY", "DUMMY")
+    assert AnthropicAPI.check_environment_variables() == []
+
+
+def test_anthropic_check_prompt_dict(temporary_data_folders, monkeypatch):
```
Comment (EdwinB12) [inline tests/apis/anthropic/test_anthropic.py:100]: Add a docstring or comment summarising what edge cases we're testing here as its quite a long function

Comment (EdwinB12) [reply tests/apis/anthropic/test_anthropic.py:100]: Actually, each case is individually well commented. ignore this.

**Review by EdwinB12 — APPROVED** [review]
Approved with just some general comments. Very impressive testing!
